service/actions/general.js

```diff
@@ -2,17 +2,28 @@
 // See License.txt for license information.
 
 import Client from 'service/client';
-import {bindClientFunc} from './helpers.js';
+import {bindClientFunc, FormattedError} from './helpers.js';
 import {GeneralTypes} from 'service/constants';
 import {getMyChannelMembers} from './channels';
 
 export function getPing() {
-    return bindClientFunc(
-        Client.getPing,
-        GeneralTypes.PING_REQUEST,
-        GeneralTypes.PING_SUCCESS,
-        GeneralTypes.PING_FAILURE
-    );
+    return async (dispatch, getState) => {
+        dispatch({type: GeneralTypes.PING_REQUEST}, getState);
+
+        let data;
+        try {
+            data = await Client.getPing();
+            if (!data.version) {
+                // successful ping but not the right return data
+                throw new FormattedError('error.network_request_failed', 'Cannot connect to the server. Please check your server URL and internet connection.');
+            }
+        } catch (error) {
+            dispatch({type: GeneralTypes.PING_FAILURE, error}, getState);
```

Comment:
@thomchop is partially right, as we don't really care in this case what's the error return from the ping request, as there is no specific errors that can occur in that enpoint so we should always return the *friendly* message